### PR TITLE
WebGLLights: Simplified hash value

### PR DESCRIPTION
**Pull request by aardgoose**
Summarise lights hash to a single sequence value and simplify renderer hash checks.
Also only truncate lights arrays when lengths have changed.

### CODE REVIEW

**Comment by mrdoob** [comment]
This looks great!

File: src/renderers/webgl/WebGLLights.js [hunk 1]
```diff
@@ -117,7 +117,8 @@ function WebGLLights() {
 			spotLength: - 1,
 			rectAreaLength: - 1,
 			hemiLength: - 1,
-			shadowsLength: - 1
+			shadowsLength: - 1,
+			value: 0
```
Comment (mrdoob) [inline src/renderers/webgl/WebGLLights.js]: To be consistent with similar code, I would name this property `version`.

Comment (aardgoose) [reply src/renderers/webgl/WebGLLights.js]: done, I wasn't happy with that name.

**Comment by mrdoob** [comment]
Definitely looks more sane 😇
Merging this soon so we can test it for the rest of the cycle.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**